actionpack/test/dispatch/mime_type_test.rb

```diff
@@ -30,67 +30,67 @@ class MimeTypeTest < ActiveSupport::TestCase
+    assert_mime_types_equal expect, parsed
   end
 
   test "parse application with trailing star" do
     accept = "application/*"
     expect = [Mime[:html], Mime[:js], Mime[:xml], Mime[:rss], Mime[:atom], Mime[:yaml], Mime[:url_encoded_form], Mime[:json], Mime[:pdf], Mime[:zip], Mime[:gzip]]
     parsed = Mime::Type.parse(accept)
-    assert_equal expect, parsed
+    assert_mime_types_equal expect, parsed
   end
 
   test "parse without q" do
     accept = "text/xml,application/xhtml+xml,text/yaml,application/xml,text/html,image/png,text/plain,application/pdf,*/*"
     expect = [Mime[:html], Mime[:xml], Mime[:yaml], Mime[:png], Mime[:text], Mime[:pdf], '*/*']
-    assert_equal expect.map(&:to_s), Mime::Type.parse(accept).map(&:to_s)
+    assert_mime_types_equal expect, Mime::Type.parse(accept)
   end
 
   test "parse with q" do
     accept = "text/xml,application/xhtml+xml,text/yaml; q=0.3,application/xml,text/html; q=0.8,image/png,text/plain; q=0.5,application/pdf,*/*; q=0.2"
     expect = [Mime[:html], Mime[:xml], Mime[:png], Mime[:pdf], Mime[:text], Mime[:yaml], '*/*']
-    assert_equal expect.map(&:to_s), Mime::Type.parse(accept).map(&:to_s)
+    assert_mime_types_equal expect, Mime::Type.parse(accept)
   end
 
   test "parse single media range with q" do
     accept = "text/html;q=0.9"
     expect = [Mime[:html]]
-    assert_equal expect, Mime::Type.parse(accept)
+    assert_mime_types_equal expect, Mime::Type.parse(accept)
   end
 
   test "parse arbitrary media type parameters" do
     accept = 'multipart/form-data; boundary="simple boundary"'
     expect = [Mime[:multipart_form]]
-    assert_equal expect, Mime::Type.parse(accept)
+    assert_mime_types_equal expect, Mime::Type.parse(accept)
   end
 
   # Accept header send with user HTTP_USER_AGENT: Sunrise/0.42j (Windows XP)
   test "parse broken acceptlines" do
     accept = "text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/*,,*/*;q=0.5"
     expect = [Mime[:html], Mime[:xml], "image/*", Mime[:text], '*/*']
-    assert_equal expect.map(&:to_s), Mime::Type.parse(accept).map(&:to_s)
+    assert_mime_types_equal expect, Mime::Type.parse(accept)
   end
 
   # Accept header send with user HTTP_USER_AGENT: Mozilla/4.0
   #  (compatible; MSIE 6.0; Windows NT 5.1; SV1; .NET CLR 1.1.4322; InfoPath.1)
   test "parse other broken acceptlines" do
     accept = "image/gif, image/x-xbitmap, image/jpeg, image/pjpeg, application/x-shockwave-flash, application/vnd.ms-excel, application/vnd.ms-powerpoint, application/msword,  , pronto/1.00.00, sslvpn/1.00.00.00, */*"
     expect = ['image/gif', 'image/x-xbitmap', 'image/jpeg','image/pjpeg', 'application/x-shockwave-flash', 'application/vnd.ms-excel', 'application/vnd.ms-powerpoint', 'application/msword', 'pronto/1.00.00', 'sslvpn/1.00.00.00', '*/*']
-    assert_equal expect.map(&:to_s), Mime::Type.parse(accept).map(&:to_s)
+    assert_mime_types_equal expect, Mime::Type.parse(accept)
```

Comment:
This test needs to check the order since the accept header is in a specific order